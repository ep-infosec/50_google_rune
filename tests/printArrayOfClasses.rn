//  Copyright 2021 Google LLC.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

class Foo(self) {
  self.bars = arrayof(Bar(self, u64))
}

class Bar:u16 (self, foo: Foo, val: u64) {
  self.val = val
  foo.bars.append(self)
}

foo = Foo()
Bar(foo, 1)
Bar(foo, 2)
Bar(foo, 3)
println foo
