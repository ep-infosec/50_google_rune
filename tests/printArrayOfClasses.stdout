{bars = [0u16, 0u16, 0u16]}
